wordpress/wp-content/plugins/memberful-wp/src/marketing_content.php

```diff
@@ -4,14 +4,31 @@
 define( 'MEMBERFUL_OPTION_DEFAULT_MARKETING_CONTENT', 'memberful_default_marketing_content' );
 
 function memberful_marketing_content( $post_id ) {
-  $memberful_marketing_content = get_post_meta( $post_id, MEMBERFUL_MARKETING_META_KEY, TRUE );
-  return apply_filters( 'memberful_marketing_content', $memberful_marketing_content );
+  $user_id = is_user_logged_in() ? get_current_user_id() : 0;
+  $restricted_posts = memberful_wp_user_disallowed_post_ids( $user_id );
+
+  if ( isset( $restricted_posts[$post_id] )) {
+    $memberful_marketing_content = get_post_meta( $post_id, MEMBERFUL_MARKETING_META_KEY, TRUE );
+    return apply_filters( 'memberful_marketing_content', $memberful_marketing_content );
+  } else {
+    $terms = memberful_terms_restricting_post( $user_id, $post_id );
+    return memberful_term_marketing_content( reset( $terms ));
```

Comment:
I'm not sure if we should call `reset()` here. If it is really needed, then we should call it in `memberful_term_marketing_content()` before we return the array.

Reply:
Ah, never mind. I've misunderstood what this should do.